### PR TITLE
Feature: interactive table for parameters

### PR DESCRIPTION
**Pull request by alashchev17**
# Feature: interactive table for parameters

## Description

- Support and rendering of a table for parameters in cmdline_ and service_ integrations
- [🔴 UI - editing command line params, a table](https://refact.fibery.io/Software_Development/Week1216-by-state-427#Task/UI---editing-command-line-params,-a-table-535)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, only code improvements)
- [ ] Documentation update

### Tool parameters table
Tool parameters table

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I have updated the documentation where necessary.

## Additional Notes

- There are some issues with validation of rather data got changed or not
- Validation for snake_case of param names is not present yet.

### CODE REVIEW

File: src/utils/isEqual.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+export function isEqual<T>(obj1: T, obj2: T): boolean {
+  if (obj1 === obj2) return true;
+
```
Comment (MarcMcIntosh) [inline src/utils/isEqual.ts:3]:
```
NaN === NaN 
```
There's probably a few more cases.

i think `lodash.isequal` is already installed :)

File: src/components/IntegrationsView/IntegrationDocker/IntegrationDocker.tsx [hunk 2]
```diff
@@ -86,11 +86,32 @@ export const IntegrationDocker: FC<IntegrationDockerProps> = ({
...
+        {/* TODO: duplicative code */}
+        {dockerData.smartlinks.length > 0 && (
+          <Flex direction="column" width="100%" gap="3" mt="2">
+            <Heading size="4" as="h4">
+              Ask AI to do it for you (experimental)
+            </Heading>
+            <Flex gap="2" align="center">
+              {dockerData.smartlinks.map((smartlink) => (
+                <SmartLink
+                  key={`docker-container-${dockerData.filter_image}`}
```
Comment (MarcMcIntosh) [inline src/components/IntegrationsView/IntegrationDocker/IntegrationDocker.tsx:104]: would this create duplicate keys?

File: src/components/IntegrationsView/IntegrationForm/IntegrationForm.tsx [hunk 3]
```diff
@@ -154,19 +173,21 @@ export const IntegrationForm: FC<IntegrationFormProps> = ({
...
-                      value={availabilityValues[key]}
-                      onChange={handleAvailabilityChange}
-                    />
-                  ),
-                )}
-            </Flex>
+            {integration.data.integr_values && (
+              <Flex gap="4" mb="3" className={styles.switchInline}>
+                {integration.data.integr_values.available &&
+                  Object.entries(integration.data.integr_values.available).map(
```
Comment (MarcMcIntosh) [inline src/components/IntegrationsView/IntegrationForm/IntegrationForm.tsx:179]: Objecy.keys would work aswell.

**Review by MarcMcIntosh — COMMENTED** [review]
small changes, 

I'll run it to see if there's any issues :)

File: src/components/IntegrationsView/IntegrationDocker/IntegrationDocker.tsx (hunk 2 above)
Comment (alashchev17) [reply src/components/IntegrationsView/IntegrationDocker/IntegrationDocker.tsx:104]: Should not do that. We will see those smartlinks only if there are no containers, but if there are some of them - we'll see different part of the UI

Comment (alashchev17) [reply src/components/IntegrationsView/IntegrationDocker/IntegrationDocker.tsx:104]: There is no overlap of those keys

File: src/components/IntegrationsView/IntegrationForm/IntegrationForm.tsx (hunk 3 above)
Comment (alashchev17) [reply src/components/IntegrationsView/IntegrationForm/IntegrationForm.tsx:184]: Done

**Review by MarcMcIntosh — APPROVED** [review]

**Pull request merged**